In [1]:
import numpy as np

In [29]:
sagunsago = [15129, 13836, 15738, 16300, 17605, 16690, 16252, 17000, 16956, 17720, 17455, 15668]

s_sorted = sorted(sagunsago)
print("사건사고 정렬:",s_sorted)

s_len = len(sagunsago)
print(s_len)
#중위수, 1~3사분위 수 
if s_len % 2 == 0:
    
    sm_1 = (s_len // 2 )-1
    sm_2 = sm_1+1
    median = (s_sorted[sm_1] + s_sorted[sm_2])/2

else:
    median = s_sorted[s_len /2]

print("중앙값:",median)


# 1사분위 수
q1_s = (s_len-1) * 0.25 
i = int(np.floor(q1_s))
f = q1_s - i

Q1 = s_sorted[i] + f * (s_sorted[i + 1]- s_sorted[i])
print("1사분위 수:", Q1)

# 3사분위 수
q3_s = (s_len-1) * 0.75
i = int(np.floor(q3_s))
f = q3_s - i

Q3_s = s_sorted[i] - f * (s_sorted[i + 1] - s_sorted[i])
print("3분위수:", Q3_s)

# 사분위수 구간
print("사분위수 구간:", np.quantile(sagunsago, 0.75)-np.quantile(sagunsago, 0.25) )

# 데이터 구간
print("데이터 구간:", np.max(sagunsago) - np.min(sagunsago))


print("분산:", np.var(sagunsago))
print("표준편차:", np.std(sagunsago))

# 구간
hist, bins = np.histogram(sagunsago, 5)
print("구간 별 데이터 수:", hist)
print("구간 경계:", bins)

# 도수분포표 구성
for i in range(0, len(hist)):
    if i + 1 < len(bins):
        tpl = "%d이상 %d미만: %d개"

    else:
        tpl = "%d이상 %d이하: %d개"

    print(tpl % (bins[i], bins[i+1], hist[i]))

    # 상대도수
    r_freq = hist/np.sum(hist)
    print("상대도수:", r_freq)


    # 누적 도수
    c_freq = []

    previous = 0

    for i in hist:
        previous += i
        c_freq.append(previous)
    
    print("누적도수:", previous)


사건사고 정렬: [13836, 15129, 15668, 15738, 16252, 16300, 16690, 16956, 17000, 17455, 17605, 17720]
12
중앙값: 16495.0
1사분위 수: 15720.5
3분위수: 16886.25
사분위수 구간: 1393.25
데이터 구간: 3884
분산: 1186603.7430555557
표준편차: 1089.3134273732037
구간 별 데이터 수: [1 1 2 3 5]
구간 경계: [13836.  14612.8 15389.6 16166.4 16943.2 17720. ]
13836이상 14612미만: 1개
상대도수: [0.08333333 0.08333333 0.16666667 0.25       0.41666667]
누적도수: 12
14612이상 15389미만: 1개
상대도수: [0.08333333 0.08333333 0.16666667 0.25       0.41666667]
누적도수: 12
15389이상 16166미만: 2개
상대도수: [0.08333333 0.08333333 0.16666667 0.25       0.41666667]
누적도수: 12
16166이상 16943미만: 3개
상대도수: [0.08333333 0.08333333 0.16666667 0.25       0.41666667]
누적도수: 12
16943이상 17720미만: 5개
상대도수: [0.08333333 0.08333333 0.16666667 0.25       0.41666667]
누적도수: 12


첫 번째로 2024년도 월별 교통사고 데이터를 통해 표준편차가 1089로 월별 발생하 사고 수의 차이가 크지 않다는 것을 확인할 수 있었습니다.
두 번째로, 9월 부터 11월, 즉 하반기에 16900건 ~ 17700건 이하의 교통사고 건수가 집중된 것을 파악할 수 있었습니다. 

In [ ]:
# 251112
왜?
정보 파악
~하는 것을 보인다.
